modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/LocalSubscription.java

```diff
@@ -90,11 +90,16 @@ public LocalSubscription(OutboundSubscription subscription, String subscriptionI
 
         this.subscription = subscription;
 
-        //We need to keep the subscriptionType in basic subscription for notification
-        if(subscription instanceof AMQPLocalSubscription) {
-            setSubscriptionType(SubscriptionType.AMQP);
-        } else if(subscription instanceof MQTTLocalSubscription) {
-            setSubscriptionType(SubscriptionType.MQTT);
+        // In case of mock subscriptions, this becomes null
+        if (null != subscription) {
+            //We need to keep the subscriptionType in basic subscription for notification
+            if(subscription instanceof AMQPLocalSubscription) {
+                setSubscriptionType(SubscriptionType.AMQP);
+            } else if(subscription instanceof MQTTLocalSubscription) {
+                setSubscriptionType(SubscriptionType.MQTT);
```

Comment:
is there a way to separate out MQTT related stuff from kernel ( may be we can do it in next release)